src/main/java/net/minecraftforge/coremod/CoreMod.java

```diff
@@ -55,32 +55,50 @@ void initialize() {
     }
 
     @SuppressWarnings("unchecked")
-    private ITransformer<?> buildCore(Map.Entry<String,ScriptObjectMirror> entry) {
+    private ITransformer<?> buildCore(Map.Entry<String,Map<String, Object>> entry) {
         final String coreName = entry.getKey();
-        final ScriptObjectMirror data = entry.getValue();
-        final ScriptObjectMirror targetData = (ScriptObjectMirror) data.getMember("target");
-        final ITransformer.TargetType targetType = ITransformer.TargetType.valueOf((String) targetData.getMember("type"));
+        final Map<String, Object> data = entry.getValue();
```

Comment:
Small note about the switch to Map<String, Object> here. The official Java API just specifies Object.
However all known engines return Map<String, Object>'s for JS interop values. This includes for array returns, the map is just index_as_number_string -> value. It's nasty but it holds true for all known engines. My bet is that Nashorn did it, and the rest followed suit to be as compatible as possible.